core/trino-main/src/main/java/io/trino/execution/UseTask.java

```diff
@@ -82,4 +96,14 @@ public ListenableFuture<Void> execute(
 
         return immediateVoidFuture();
     }
+
+    private boolean hasCatalogAccess(SecurityContext securityContext, String catalog)
+    {
+        return !accessControl.filterCatalogs(securityContext, ImmutableSet.of(catalog)).isEmpty();
```

Comment:
Maybe we should make `io.trino.security.AccessControlManager#checkCanAccessCatalog` part of the `AccessControl` interface?

Reply:
You would need to have the same for schema too then.

Reply:
Yeah, I know.

Reply:
I can't resolve threads in this repository for some reason, but if I could, I would this one ;)